zerver/lib/outgoing_webhook.py

```diff
@@ -32,6 +34,9 @@ def __init__(self, token: str, user_profile: UserProfile, service_name: str) ->
         self.user_profile: UserProfile = user_profile
         self.service_name: str = service_name
         self.session: Session = Session()
+        self.session.request = functools.partial(
```

Comment:
Grr, mypy doesn't like this.  Short of [subclassing the adapter](https://github.com/psf/requests/issues/3341#issuecomment-226763313), there's no cleaner way to do this.  I guess we'll subclass the adapter, then. :/